src/NuGet.Clients/NuGet.VisualStudio.Implementation/SolutionExplorer/Items/PackageReferenceItem.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.VS;
+using Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections;
+
+using NuGet.VisualStudio.Implementation.Resources;
+using NuGet.VisualStudio.SolutionExplorer.Models;
+
+namespace NuGet.VisualStudio.SolutionExplorer
+{
+    /// <summary>
+    /// Backing object for transitive package reference nodes in the dependencies tree.
+    /// </summary>
+    internal sealed class PackageReferenceItem : RelatableItemBase
+    {
+        public AssetsFileTarget Target { get; private set; }
+        public AssetsFileTargetLibrary Library { get; private set; }
+
+        public PackageReferenceItem(AssetsFileTarget target, AssetsFileTargetLibrary library)
+            : base(GetCaption(library))
+        {
+            Library = library;
+            Target = target;
+        }
+
+        internal bool TryUpdateState(AssetsFileTarget target, AssetsFileTargetLibrary library)
+        {
+            if (ReferenceEquals(Target, target) && ReferenceEquals(Library, library))
+            {
+                return false;
+            }
+
+            Target = target;
+            Library = library;
+            Text = GetCaption(library);
+            return true;
+        }
+
+        private static string GetCaption(AssetsFileTargetLibrary library) => $"{library.Name} ({library.Version})";
+
+        public override object Identity => Library.Name;
+
+        public override int Priority => AttachedItemPriority.Package;
+
+        public override ImageMoniker IconMoniker => ManagedImageMonikers.NuGetGrey;
+
+        protected override bool TryGetProjectNode(IProjectTree targetRootNode, IRelatableItem item, [NotNullWhen(returnValue: true)] out IProjectTree? projectTree)
```

Comment:
The base has docs. VS shows them on overrides. If I duplicate them here (and onto the many other item subclasses), they'll just fall out of date. I encourage you to review this in VS rather than on GitHub.

https://github.com/dotnet/project-system/blob/1587349ca69b35ce320f7af8f264e89ce12bdea6/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/IRelatableItem.cs#L71-L86

Reply:
I was reviewing in VS and it showed nothing for me, weird. 
That'd helpful though, thanks.

Reply:
Yeah I realised after writing that comment that this API was missing a hint, hence this PR: https://github.com/dotnet/project-system/pull/6251